### PR TITLE
Aws ripe profile

### PR DESCRIPTION
**Pull request by maggarwal13**

### CODE REVIEW

File: whois-commons/src/main/java/net/ripe/db/whois/common/WhoisAWSPropertyResolver.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.stream.Collectors;
+
+@Configuration
+@Profile(WhoisProfile.AWS_DEPLOYED)
+@PropertySource(value = "classpath:version.properties", ignoreResourceNotFound = true)
+public class WhoisAWSPropertyResolver {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WhoisAWSPropertyResolver.class);
+
+    //TODO: get region to deploy from gitlab
```
Comment (sbusk) [inline whois-commons/src/main/java/net/ripe/db/whois/common/WhoisAWSPropertyResolver.java:30]: Should be provided as a environment variable (hopefully automatically).

File: whois-commons/src/main/java/net/ripe/db/whois/common/WhoisAWSPropertyResolver.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        Properties properties = new Properties();
+        properties.putAll( getParametersByEnvAndApp("/whois/", true));
+
+        PropertySourcesPlaceholderConfigurer propertySourceConfig = new PropertySourcesPlaceholderConfigurer();
+        propertySourceConfig.setProperties(properties);
+
+        return propertySourceConfig;
+    }
+
+    public static Map<String, Object> getParametersByEnvAndApp(final String path, final boolean encryption) {
```
Comment (sbusk) [inline whois-commons/src/main/java/net/ripe/db/whois/common/WhoisAWSPropertyResolver.java:46]: If encryption is only ever true, why pass it as an argument?

File: whois-commons/src/main/java/net/ripe/db/whois/common/WhoisAWSPropertyResolver.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        final AWSSimpleSystemsManagement awsSimpleSystemsManagement = AWSSimpleSystemsManagementClient.builder()
+                .withRegion(REGION).build();
+
+        final GetParametersByPathRequest getParametersByPathRequest = new GetParametersByPathRequest().withPath(path)
```
Comment (sbusk) [inline whois-commons/src/main/java/net/ripe/db/whois/common/WhoisAWSPropertyResolver.java:50]: Nitpicking here but .withPath(...) should be on next line.

File: whois-commons/src/main/java/net/ripe/db/whois/common/WhoisAWSPropertyResolver.java (hunk 2 above)
Comment (maggarwal13) [reply whois-commons/src/main/java/net/ripe/db/whois/common/WhoisAWSPropertyResolver.java:46]: ok, I will make it default true always, as it needs to decrypt the passwords

File: whois-db/pom.xml [hunk 4]
```diff
@@ -171,6 +171,8 @@
                         <jvmFlags>
                             <jvmFlag>-Dwhois.config=/export/services/whois/properties</jvmFlag>
                             <jvmFlag>-Dlog4j.configurationFile=file:/export/services/whois/log4j2.xml</jvmFlag>
+                            <!-- TODO: get profile from gitlab -->
```
Comment (sbusk) [inline whois-db/pom.xml:174]: We'll put this in Terraform.

**Review by sbusk — APPROVED** [review]

**Pull request merged**